From f6874a2f7de30b6eeefb2d509b895c58b93da8f7 Mon Sep 17 00:00:00 2001
From: Sasi Kumar Maddineni <quic_sasikuma@quicinc.com>
Date: Wed, 29 Oct 2025 09:38:17 +0530
Subject: [PATCH] libziparchive: fix missing fixed-width integer types in
 zip_writer.h

Compilation fails due to undefined types 'uint16_t' and 'uint32_t' in
zip_writer.h. These types are used in struct members but are not recognized
because the required header <cstdint> is not included. This results in
errors such as:
  error: 'uint16_t' does not name a type
  error: 'uint32_t' does not name a type

Added '#include <cstdint>' to zip_writer.h to ensure fixed-width integer
types are properly defined.

Upstream-Status: Pending

Signed-off-by: Sasi Kumar Maddineni <quic_sasikuma@quicinc.com>
---
 system/core/libziparchive/include/ziparchive/zip_writer.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/system/core/libziparchive/include/ziparchive/zip_writer.h b/system/core/libziparchive/include/ziparchive/zip_writer.h
index d68683df..7a05e349 100644
--- a/system/core/libziparchive/include/ziparchive/zip_writer.h
+++ b/system/core/libziparchive/include/ziparchive/zip_writer.h
@@ -18,6 +18,7 @@
 
 #include <cstdio>
 #include <ctime>
+#include <cstdint>
 
 #include <gtest/gtest_prod.h>
 #include <memory>
-- 
2.34.1

